sympy/solvers/solvers.py

```diff
@@ -2507,15 +2506,14 @@ def det_perm(M):
     args = []
     s = True
     n = M.rows
-    try:
-        list = M._mat
-    except AttributeError:
-        list = flatten(M.tolist())
+    list_ = getattr(M, '_mat', None)
```

Comment:
`list_ = getattr(M, '_mat', flatten(M.tolist()))`

(assuming that the default value is not evaluated until it is possibly required)

Reply:
No, all arguments are evaluated before a function is called.

Reply:
OK, good to know!